py/llama_cloud_services/beta/agent_data/schema.py

```diff
@@ -222,12 +222,16 @@ def parse_extracted_field_metadata(
     return {
         k: _parse_extracted_field_metadata_recursive(v)
         for k, v in field_metadata.items()
-        if k not in _METADATA_FIELDS_SIBLING_TO_LEAF
-        and k not in _ADDITIONAL_ROOT_METADATA_FIELDS
+        if not _is_reasoning_field(k, v) and k not in _ADDITIONAL_ROOT_METADATA_FIELDS
```

Comment:
Do we need to have a citation related check too? If it's easier to distinguish metadata fields by adding a user provided suffix, let me know.

Reply:
The citation might need one too, but I think it’s more difficult to create that scenario. The reasoning field can show up as a sibling to extracted fields in some cases. This currently has the behavior of copying the reasoning field down to leaves when present. Erroneously doing so when the reasoning is a schema provided field (with citations, etc) led to the error

Reply:
Configuring the reasoning output with a suffix/prefix does make sense to avoid name collisions, although that might actually make this parsing more complex